### PR TITLE
[SMAGENT-3306] Add --image_pull_secret flag to IBM installer

### PR DESCRIPTION
**Pull request by fuster92**
Following the changes done to the default installer by @alblasco in https://github.com/draios/agent/pull/2880 I'm porting them to
the IBM installer.

### CODE REVIEW

File: agent_deploy/IBMCloud-Kubernetes-Service/install-agent-k8s.sh [hunk 1]
```diff
@@ -383,6 +384,11 @@ function install_k8s_agent {
         AGENT_NAMES="agent-slim agent-kmodule"
     fi
 
+	if [ ! -z "$IMAGE_PULL_SECRET" ]; then
+		sed -i.bak -e "s|#imagePullSecrets:|imagePullSecrets:|" $DAEMONSET_FILE
+		sed -i.bak -e "s|#- name: secret-name|- name: ${IMAGE_PULL_SECRET}|" $DAEMONSET_FILE
```
Comment (irozzo-1A) [inline agent_deploy/IBMCloud-Kubernetes-Service/install-agent-k8s.sh:389]: I think you should be able to use a single command:
```suggestion
		sed -i.bak -e "s|#imagePullSecrets:|imagePullSecrets:|" \
			-e "s|#- name: secret-name|- name: ${IMAGE_PULL_SECRET}|" \
			$DAEMONSET_FILE
```

**Review by alblasco — DISMISSED** [review]
LGTM

**Review by papajulio — DISMISSED** [review]

**Comment by fuster92** [comment]
Changed the way to add the image pull secret due to it being overwritten, now it checks whether a secret already exists or not.

**Review by alblasco — DISMISSED** [review]

File: agent_deploy/IBMCloud-Kubernetes-Service/install-agent-k8s.sh [hunk 2]
```diff
@@ -742,6 +757,15 @@ case ${key} in
...
         ;;
+    -ips|--image_pull_secret)
+			if is_valid_value "${2}"; then
+				IMAGE_PULL_SECRET="${2}"
+			else
+				echo "ERROR: no value provided for image pull secret option, use -h | --help for $(basename ${0}) Usage"
+				exit 1
+			fi
+			shift
+		;;
```
Comment (marojor) [inline agent_deploy/IBMCloud-Kubernetes-Service/install-agent-k8s.sh:768]: Indentation looks inconsistent

File: agent_deploy/IBMCloud-Kubernetes-Service/install-agent-k8s.sh [hunk 3]
```diff
@@ -434,6 +436,18 @@ function install_k8s_agent {
...
+        if grep -E '^\s*imagePullSecrets:' $DAEMONSET_FILE ; then
+            INDENT=$(grep 'containers' $DAEMONSET_FILE | sed 's/\( *\).*/\1/')
+            echo "$INDENT- name: ${IMAGE_PULL_SECRET}" >> $DAEMONSET_FILE
+        else
+
+            sed -i.bak -e "s|#imagePullSecrets:|imagePullSecrets:|" \
+                -e "s|#- name: secret-name|- name: ${IMAGE_PULL_SECRET}|" \
+                $DAEMONSET_FILE
+        fi
+	fi
```
Comment (marojor) [inline agent_deploy/IBMCloud-Kubernetes-Service/install-agent-k8s.sh:449]: Fix indentation please.

File: agent_deploy/IBMCloud-Kubernetes-Service/install-agent-k8s.sh [hunk 4]
```diff
@@ -383,6 +384,7 @@ function install_k8s_agent {
         AGENT_NAMES="agent-slim agent-kmodule"
     fi
 
+
```
Comment (marojor) [inline agent_deploy/IBMCloud-Kubernetes-Service/install-agent-k8s.sh:387]: Is this new empty line to comply with current social distance? :trollface:

**Review by alblasco — APPROVED** [review]

**Review by papajulio — APPROVED** [review]

**Review by marojor — APPROVED** [review]

**Review by alblasco — APPROVED** [review]

**Review by balinaveen — APPROVED** [review]

**Pull request merged**